lib/rspec-puppet/adapters.rb

```diff
@@ -0,0 +1,107 @@
+module RSpec::Puppet
+  module Adapters
+
+    class Base
+      def setup_puppet(example_group)
+        settings_map.each do |puppet_setting, rspec_setting|
+          set_setting(example_group, puppet_setting, rspec_setting)
+        end
+      end
+
+      def set_setting(example_group, puppet_setting, rspec_setting)
+        if example_group.respond_to?(rspec_setting)
+          value = example_group.send(rspec_setting)
+        else
+          value = RSpec.configuration.send(rspec_setting)
+        end
+        begin
+          Puppet[puppet_setting] = value
+        rescue ArgumentError
+          # TODO: this silently swallows errors when applying settings that the current
```

Comment:
Kinda. This could cause a large number of warnings to be emitted; I was just (mostly) preserving the existing behavior so this isn't a regression. Should we fix this here or should we spin this out into another issue?